### ISSUE TITLE
invalid html for citation nodes with no siblings

### ISSUE THREAD

**Issue by mcmtroffaes** [sphinx-doc/sphinx#10784]
### Describe the bug

When citations appear in a parent that has no other siblings besides this single citation, a spurious ``</div>`` is generated resulting in invalid html code. It appears that this is because the opening ``<div role="list" class="citation-list">`` is not being generated. I'm unsure whether this is a bug in Sphinx or in docutils, but I've tracked it to this docutils failing to add an opening div for the citation list here:

https://github.com/docutils/docutils/blob/master/docutils/docutils/writers/_html_base.py#L632

It gets then closed here leading to the extra tag:

https://github.com/docutils/docutils/blob/master/docutils/docutils/writers/_html_base.py#L640

Basically, this is due to odd behaviour of previous_sibling: if index relative to parent is zero and there are no other siblings, it returns the node itself instead of None. I don't know if this is a docutils bug or not.

### How to Reproduce

Let sphinx generate html for the following one-line document:

```
.. [CIT2022] A citation.
```

### Expected behavior

Valid html code.

### Your project

https://github.com/mcmtroffaes/sphinx-citation-invalid-html

### OS

Windows 11

### Python version

3.10

### Sphinx version

5.1.1

### Additional context

Originally reported here: https://github.com/mcmtroffaes/sphinxcontrib-bibtex/issues/309

**Comment by mcmtroffaes** [comment]
Docutils patch submitted to sourceforge: https://sourceforge.net/p/docutils/patches/195/

**Comment by Abramo-Bagnara** [comment]
@AA-Turner This problem is a show stopper for users needing a valid xhtml output, there is a reason why it has not been applied to recent releases?

**Comment by AA-Turner** [comment]
@Abramo-Bagnara all contributors to Sphinx and Docutils are volunteers, if you would like a fix quicker a PR is always welcome. Note there are nearly a thousand open issues, all which may deserve attention.

A

**Comment by Abramo-Bagnara** [comment]
I didn't mean to press you to do something you have not the energies to do or it is not among your priorities. My question was only about possible objections to inclusion. From your answer I deduce that no objection is pending, thus I'll submit a PR now.
Sorry for the misunderstanding.

**Comment by mcmtroffaes** [comment]
@Abramo-Bagnara It's most likely not a sphinx issue, but a docutils issue (as far as I can tell at least), with a patch awaiting merge by docutils (they aren't on github so don't do PRs). See https://github.com/sphinx-doc/sphinx/issues/10784#issuecomment-1226971018 If it is important to you, you can use the patch, and optionally report your experience with it on the docutils issue tracker (esp. if it wouldn't be working in some cases).

**Comment by gmilde** [comment]
This is fixed in [Docutils 0.20](https://sourceforge.net/p/docutils/patches/195/#0bf6) (2023-05-04).

I suggest closing this issue.
